### PR TITLE
Fix missing userId in LOGIN_ERROR event for permanent lockout with se…

### LINKED ISSUE #35732: Missing userId in LOGIN_ERROR event for permanent lockout
### Area

authentication

### Describe the bug

When brute force protection is enabled and configured for permanent lockout, using separate "Username" and "Password" form authenticators causes the `LOGIN_ERROR` event log to omit the userId field. This occurs because `event.user(user)` is never called before the `AuthenticationFlowException` is thrown in the permanent lockout scenario. The attached PR ensures the user is bound to the event before the exception is raised, restoring the expected `userId` in the event log.

### Version

26.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

The `LOGIN_ERROR` event should include the userId field, even under permanent lockout conditions.

### Actual behavior

The `LOGIN_ERROR` event does not include the userId field under permanent lockout, and throws an exception :

```
2024-10-17 10:43:38,199 WARN  [org.keycloak.events] (executor-thread-26) type="LOGIN_ERROR", realmId="15b3a838-647e-4f17-9a00-7116f39e0b62", realmName="test", clientId="account-console", userId="4e2fbbbc-289d-435e-9fe7-4f1719226bb0", ipAddress="192.168.65.1", error="invalid_user_credentials", auth_method="openid-connect", auth_type="code", redirect_uri="http://localhost:8080/realms/test/account", code_id="43d6b496-2c61-4b6d-9689-a318dbcaefc2"
2024-10-17 10:43:41,083 WARN  [org.keycloak.events] (executor-thread-26) type="LOGIN_ERROR", realmId="15b3a838-647e-4f17-9a00-7116f39e0b62", realmName="test", clientId="account-console", userId="4e2fbbbc-289d-435e-9fe7-4f1719226bb0", ipAddress="192.168.65.1", error="user_temporarily_disabled", auth_method="openid-connect", auth_type="code", redirect_uri="http://localhost:8080/realms/test/account", code_id="43d6b496-2c61-4b6d-9689-a318dbcaefc2"
2024-10-17 10:44:50,407 WARN  [org.keycloak.events] (executor-thread-27) type="LOGIN_ERROR", realmId="15b3a838-647e-4f17-9a00-7116f39e0b62", realmName="test", clientId="account-console", userId="4e2fbbbc-289d-435e-9fe7-4f1719226bb0", ipAddress="192.168.65.1", error="invalid_user_credentials", auth_method="openid-connect", auth_type="code", redirect_uri="http://localhost:8080/realms/test/account", code_id="43d6b496-2c61-4b6d-9689-a318dbcaefc2"
2024-10-17 10:44:52,322 WARN  [org.keycloak.services] (executor-thread-28) KC-SERVICES0013: Failed authentication: org.keycloak.authentication.AuthenticationFlowException
        at org.keycloak.authentication.AuthenticationProcessor.validateUser(AuthenticationProcessor.java:1167)
        at org.keycloak.authentication.AuthenticationProcessor.authenticationAction(AuthenticationProcessor.java:1017)
        at org.keycloak.services.resources.LoginActionsService.processFlow(LoginActionsService.java:378)
        at org.keycloak.services.resources.LoginActionsService.processAuthentication(LoginActionsService.java:349)
        at org.keycloak.services.resources.LoginActionsService.authenticate(LoginActionsService.java:341)
        at org.keycloak.services.resources.LoginActionsService.authenticateForm(LoginActionsService.java:407)
        at org.keycloak.services.resources.LoginActionsService$quarkusrestinvoker$authenticateForm_13d428b70e62601585246c086a52f40868a3208e.invoke(Unknown Source)
        at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)

2024-10-17 10:44:52,326 WARN  [org.keycloak.events] (executor-thread-28) type="LOGIN_ERROR", realmId="15b3a838-647e-4f17-9a00-7116f39e0b62", realmName="test", clientId="account-console", userId="null", ipAddress="192.168.65.1", error="user_disabled", code_id="43d6b496-2c61-4b6d-9689-a318dbcaefc2"

```
        
### How to Reproduce?

In Keycloak, enable brute force protection on a realm and configure "Lockout permanently" or "Lockout permanently after temporary lockout"
Modify the browser authentication flow to use separate "Username" and "Password" form authenticators instead of a single "Username and Password" authenticator.
Attempt multiple failed login attempts to trigger a permanent lockout.
Inspect the `LOGIN_ERROR` events in the Admin Console or event logs. Notice that the `userId` field is missing for the permanent lockout event.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by alaa60**
Fix #35732

### CODE REVIEW

**Review by rmartinc — COMMENTED** [review]
Thanks @alaa60 for the PR!

Maybe we want to also add a test for this, but I think it's not going to be easy. You don't need brute force enabled. It's just add the username the first time with the user enabled, in the password page disable the user and the error should be there.

@mposolda Do we prepare a test for this? Or is it enough just like this?

File: services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java [hunk 1]
```diff
@@ -1177,7 +1177,10 @@ public Response finishAuthentication(LoginProtocol protocol) {
 
     public void validateUser(UserModel authenticatedUser) {
         if (authenticatedUser == null) return;
-        if (!authenticatedUser.isEnabled()) throw new AuthenticationFlowException(AuthenticationFlowError.USER_DISABLED);
+        if (!authenticatedUser.isEnabled()) {
+            event.user(authenticatedUser);
```
[the remark this answers is not in the document]
Comment (mposolda) [reply services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java:1181]: +1

**Comment by rmartinc** [comment]
Thanks @mposolda! Then @alaa60, please just add the detail for the username and is ready to go.

**Comment by alaa60** [comment]
Thanks @rmartinc for the feedback. I've added `event.detail(Details.USERNAME, authenticatedUser.getUsername());`.

**Review by rmartinc — APPROVED** [review]
Thanks @alaa60! LGTM!

**Review by mposolda — APPROVED** [review]
@alaa60 @rmartinc Thanks!

**Pull request merged**